source/slang/ir.cpp

```diff
@@ -3737,8 +3794,46 @@ namespace Slang
         context->shared = &sharedContextStorage;
         context->builder = &sharedContextStorage.builderStorage;
         context->target = target;
-
-
+        Substitutions * curTailSubst = nullptr;
```

Comment:
This is a very minor point, but this is the kind of case where I abuse a pointer-to-a-pointer to store the "link" for building a singly linked list. E.g.

```
// at the start
RefPtr<Substitutions>* substLink = &context->subst;

// later, when adding a link to the chain:
RefPtr<Substitutions> paramSubst = new ...;
*substLink = paramSubst;
substLink = &paramSubst->outer;
```

That idiom makes it a bit simpler to avoid special-casing for the start of the list, because the "link" variable always points at the last NULL pointer in the list (which for an empty list is the pointer that would point to the first node).